### ISSUE TITLE
Get-AzureRmRoleAssignment not returning names of custom roles

### ISSUE THREAD

**Issue by aaron-0** [Azure/azure-powershell#4564]
### Cmdlet(s)
Get-AzureRmRoleAssignment -ObjectId $id

### PowerShell Version
5.1.15063.502

### Module Version
4.3.1

### OS Version
10.0.15063.502

### Description
Get-AzureRmRoleAssignment -ObjectId $id doesn't return a value for RoleDefinitionName on custom role assigments:
RoleAssignmentId   : /subscriptions/###/providers/Microsoft.Authorization/roleAssignments/###
Scope              : /subscriptions/###
DisplayName        : My_AD_Group
SignInName         :
RoleDefinitionName :
RoleDefinitionId   : a5b...
ObjectId           : a4d...
ObjectType         : Group
The role is a custom role created using New-AzureRmRoleDefinition -InputFile "MyRole.json"

I noticed this issue because I have a script that removes subscription permissions first and then adds the new set of permissions that worked up until recently.

Now the following line of my script gives an error, but only for custom roles: Get-AzureRmRoleAssignment -ObjectId $objectId | Remove-AzureRmRoleAssignment

```
Remove-AzureRmRoleAssignment : Cannot validate argument on parameter 'RoleDefinitionName'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
At C:\src\AssignResourcePermissions.ps1:10 char:53
+ ... reRmRoleAssignment -ObjectId $objectId | Remove-AzureRmRoleAssignment
+                                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (Microsoft.Azure...SRoleAssignment:PSObject) [Remove-AzureRmRoleAssignment], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationError,Microsoft.Azure.Commands.Resources.RemoveAzureRoleAssignmentCommand
```

**Comment by cormacpayne** [comment]
@darshanhs90 would you mind looking at this issue? The suggested fix is the following:

1) Make sure the `RoleDefinitionName` is provided when a user gets the role assignment
2) Implement a new parameter set for `Remove-AzureRmRoleAssignment` that takes an `InputObject` of type `PSRoleAssignment` by value so users are enabled in the above piping scenario.

Please feel free to reach out to us if you have any questions on how this should be done.

**Comment by darshanhs90** [comment]
Sure will look into this and get back to you incase of any queries

**Comment by darshanhs90** [comment]
from what i checked,we do provide the roledefinitionname in the get call,irrespective of it being a custom RD or Builtin RD
PS E:\posh code> New-AzureRmRoleAssignment -ObjectId "43440d39-6989-44d5-8129-091277366972" -RoleDefinitionName $roledefname -Scope $scope


RoleAssignmentId   : /subscriptions/4004a9fd-d58e-48dc-aeb2-4a4aec58606f/ResourceGroups/rbactest/providers/Microsoft.Authorization/roleAssignments/9a557d14-bae2-44d9-8770-7c277809c447
Scope              : /subscriptions/4004a9fd-d58e-48dc-aeb2-4a4aec58606f/ResourceGroups/rbactest
DisplayName        : testapp
SignInName         :
RoleDefinitionName : RoleWithSubscriptionScope_81d33e59ce2f4fa6b98857e25467858d
RoleDefinitionId   : 81d33e59-ce2f-4fa6-b988-57e25467858d
ObjectId           : 43440d39-6989-44d5-8129-091277366972
ObjectType         : ServicePrincipal



PS E:\posh code> Get-AzureRmRoleAssignment -ObjectId "43440d39-6989-44d5-8129-091277366972"


RoleAssignmentId   : /subscriptions/4004a9fd-d58e-48dc-aeb2-4a4aec58606f/ResourceGroups/rbactest/providers/Microsoft.Authorization/roleAssignments/9a557d14-bae2-44d9-8770-7c277809c447
Scope              : /subscriptions/4004a9fd-d58e-48dc-aeb2-4a4aec58606f/ResourceGroups/rbactest
DisplayName        : testapp
SignInName         :
RoleDefinitionName : RoleWithSubscriptionScope_81d33e59ce2f4fa6b98857e25467858d
RoleDefinitionId   : 81d33e59-ce2f-4fa6-b988-57e25467858d
ObjectId           : 43440d39-6989-44d5-8129-091277366972
ObjectType         : ServicePrincipal



PS E:\posh code> Get-AzureRmRoleDefinition -Name "RoleWithSubscriptionScope_81d33e59ce2f4fa6b98857e25467858d"


Name             : RoleWithSubscriptionScope_81d33e59ce2f4fa6b98857e25467858d
Id               : 81d33e59-ce2f-4fa6-b988-57e25467858d
IsCustom         : True
Description      : Test Role definition --RoleWithSubscriptionScope
Actions          : {Microsoft.Authorization/*/read, */action}
NotActions       : {}
AssignableScopes : {/subscriptions/4004a9fd-d58e-48dc-aeb2-4a4aec58606f/ResourceGroups/rbactest}

**Comment by darshanhs90** [comment]
Create a PR for remove-azurermroleassignment modification
https://github.com/Azure/azure-powershell/pull/4575

**Comment by aaron-0** [comment]
If it's returning correctly for you is there any further investigation you would like me to do on my end?

**Comment by darshanhs90** [comment]
@aaron-0 the best bet is to run the command in debug mode and provide us with the stacktrace. We can find out if there was any issue for this case alone or not.

**Comment by aaron-0** [comment]
@darshanhs90 I have emailed you the full debug output as it contains identifying information

**Comment by iseldur** [comment]
We have the same issue that Get-AzureRMRoleAssignment did not properly return a RoleDefinitionName for RoleAssignments where a custom role is used for the assignments. 
We did a analyse and determined that version 4.2.1 of AzureRM still works. All the versions which are released after are not working including the latest version 5.0.1. 
We ran Get-AzureRMRoleAssignment(5.0.1) with debug option and saw in the response that the customroles were not included. 
The same procedure we repeated for 4.2.1 and estathablished that the custom roles were included. Maybe that could be the reason for this issue. Hope this helps. 
Like to know when the issue will be resolved. ;-)
Using windows 10 with powershell 5.1.14393.1770.
